token-metadata/program/src/processor/metadata/print.rs

```diff
@@ -0,0 +1,194 @@
+    let edition_mint_authority_info = ctx.accounts.edition_mint_authority_info;
+    let edition_token_record_info = ctx.accounts.edition_token_record_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_edition_info = ctx.accounts.master_edition_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_marker_pda_info = ctx.accounts.edition_marker_pda_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let payer_info = ctx.accounts.payer_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_token_account_owner_info = ctx.accounts.master_token_account_owner_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_token_account_info = ctx.accounts.master_token_account_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_metadata_info = ctx.accounts.master_metadata_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let update_authority_info = ctx.accounts.update_authority_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let token_program = ctx.accounts.token_program_info;
+    let ata_program = ctx.accounts.ata_program_info;
+    let sysvar_instructions = ctx.accounts.sysvar_instructions_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let system_program = ctx.accounts.system_program_info;
+
+    // If the edition token account isn't already initialized, create it.
+    // If it does exist, validate it.
+    if edition_token_account_info.data_is_empty() {
+        // if the token account is empty, we will initialize a new one but it must
+        // be an ATA account
+        assert_derivation(
+            &spl_associated_token_account::id(),
+            edition_token_account_info,
+            &[
+                edition_token_account_owner_info.key.as_ref(),
+                spl_token::id().as_ref(),
+                edition_mint_info.key.as_ref(),
+            ],
+        )?;
+
+        // creating the associated token account
+        invoke(
+            &spl_associated_token_account::instruction::create_associated_token_account(
+                ctx.accounts.payer_info.key,
+                edition_token_account_owner_info.key,
+                edition_mint_info.key,
+                &spl_token::id(),
+            ),
+            &[
+                payer_info.clone(),
+                edition_token_account_owner_info.clone(),
+                edition_mint_info.clone(),
+                edition_token_account_info.clone(),
+            ],
+        )?;
+    } else {
+        assert_owned_by(edition_token_account_info, &spl_token::id())?;
+    }
+
```

Comment:
Do we need to assert that the `edition_token_account_info` is initialized?

Reply:
I think this is done on `process_mint_new_edition_from_master_edition_via_token_logic`, but I will leave the comment here to be sure.

Reply:
It wasn't checked so added!